### ISSUE TITLE
5-Dimension concat on TILE_LAYOUT inputs

### ISSUE THREAD

**Issue by LPanosTT** [tenstorrent/tt-metal#16217]
The following test fails:

```
def test_5d_concat(device):

    torch_input_tensors = [torch.rand(1, 32, 56, 56, 1, dtype=torch.bfloat16) for _ in range(3)]
    torch_result = torch.cat(torch_input_tensors, dim=-1)

    ttnn_input_tensors = [ttnn.from_torch(x, layout=ttnn.TILE_LAYOUT, device=device) for x in torch_input_tensors]
    ttnn_result = ttnn.concat(ttnn_input_tensors, dim=-1) # throws here
    ttnn_result = ttnn.to_torch(ttnn_result)
    assert_with_pcc(torch_result, ttnn_result, 0.9999)
```
With error: `Tensor rank is greater than 4`

Performing the test with a ROW_MAJOR_LAYOUT inputs instead, and then attempting to convert the result to TILE _LAYOUT fails on the subsequent `ttnn.to_layout` call. Test:

```
def test_5d_concat(device):

    torch_input_tensors = [torch.rand(1, 32, 56, 56, 1, dtype=torch.bfloat16) for _ in range(3)]
    torch_result = torch.cat(torch_input_tensors, dim=-1)

    ttnn_input_tensors = [ttnn.from_torch(x, layout=ttnn.ROW_MAJOR_LAYOUT, device=device) for x in torch_input_tensors]
    ttnn_result = ttnn.concat(ttnn_input_tensors, dim=-1)
    ttnn_result = ttnn.to_layout(ttnn_result, ttnn.TILE_LAYOUT) # throws here
    ttnn_result = ttnn.to_torch(ttnn_result)
    assert_with_pcc(torch_result, ttnn_result, 0.9999)
```

With error:

```
E       RuntimeError: TT_FATAL @ /localdev/lpanos/tt-metal/ttnn/cpp/ttnn/operations/data_movement/reshape_view/reshape.cpp:383: false
E       info:
E       Attempting to reshape between two shapes with different volumes
```

If we are to perform eltwise operations after the concatenation, its result must be in TILE_LAYOUT.

**Comment by ntarafdar** [comment]
@LPanosTT can you put a priority on this

**Comment by jvegaTT** [comment]
Update on this, @nardoTT has built a fix for the workaround of using row major and then converting to tile. She is currently making a PR for it to unblock your team.

**Comment by ntarafdar** [comment]
Thanks @jvegaTT and @nardoTT . @LPanosTT we can downgrade this from a P0  once @nardoTT links you the PR?

**Comment by LPanosTT** [comment]
@ntarafdar Sure that works 👍

**Comment by nardoTT** [comment]
Looking at this issue again, it seems like the changes in this PR: https://github.com/tenstorrent/tt-metal/pull/16299 are enough to solve the case with row major layout. The extra code is part of the fix for the other case.

**Comment by nardoTT** [comment]
The fix for both cases is in nardo/concat branch if it is needed in the meantime. I am just waiting to another PR to get merged to create one for this. The row major layout case is working for all dimensions and shapes. The tile layout case works for dimensions up to 6D. Dims starting from 7D are still buggy, I will create another issue for it.

**Comment by jvegaTT** [comment]
Nour has merged in the referenced PR (https://github.com/tenstorrent/tt-metal/pull/16299) which means the row major case should now work on main

**Comment by nardoTT** [comment]
I have also created the PR for the TILE_LAYOUT fix (https://github.com/tenstorrent/tt-metal/pull/16453)
